### PR TITLE
[Ingest Processor] Add `ignore_missing` param to the `uri_parts` ingest processor.

### PR DESCRIPTION
**Pull request by afoucret**
### Summary

Add `ignore_missing` param to the `uri_parts`

The default value is `false` so the behavior is unchanged when the param is not specified.

### Context

It may seems a useless enhancement because we are usually implementing the same behavior using a conditional:

```json
{
  "uri_parts": {
    "field": "page_url",
    "if": "ctx.page_url != null"
  }
}
```
It becomes tricky when using a foreach processor like in the following example:

```json
{
  "foreach": {
    "field": "items",
    "processor": {
       "uri_parts": {
         "field": "_ingest._value.page_url"
       }
    }
  }
}
```

In this situation there is no way to write a working condition. As mentioned in the #60470, it seems to be a non trivial task to modify the context to support this use case.

Adding the `ignore_missing` param seems a good workaround to unblock an enterprise search use case while waiting for a more complete solution #60470 using conditionals. The foreach processor can become:

```json
{
  "foreach": {
    "field": "items",
    "processor": {
       "uri_parts": {
         "field": "_ingest._value.page_url",
         "ignore_missing": true
       }
    }
  }
}
```

### CODE REVIEW

File: modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/UriPartsProcessor.java [hunk 1]
```diff
@@ -56,8 +67,11 @@ public boolean getKeepOriginal() {
 
     @Override
     public IngestDocument execute(IngestDocument ingestDocument) throws Exception {
-        String value = ingestDocument.getFieldValue(field, String.class);
+        String value = ingestDocument.getFieldValue(field, String.class, ignoreMissing);
 
+        if (Objects.isNull(value)) {
```
Comment (joegallo) [inline modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/UriPartsProcessor.java:72]: A nit, but `value == null` is better in this context -- `Objects.isNull` is for passing around method references, not for, like, calling.

Comment (afoucret) [reply modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/UriPartsProcessor.java:72]:
```suggestion
        if (null == value) {
```

Comment (afoucret) [reply modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/UriPartsProcessor.java]: Hey @joegallo, it is fixed.
Thank you,

**Review by joegallo — CHANGES_REQUESTED** [review]
Mostly looks okay to me, but I think you've changed the semantics of how it handles a null value in the default case (i.e. where `ignore_missing` is `false`). I added a failing test (the equivalent of which passes on `main`).

File: modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/UriPartsProcessor.java [hunk 2]
```diff
@@ -165,7 +179,8 @@ public UriPartsProcessor create(
             String targetField = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "target_field", "url");
             boolean removeIfSuccessful = ConfigurationUtils.readBooleanProperty(TYPE, processorTag, config, "remove_if_successful", false);
             boolean keepOriginal = ConfigurationUtils.readBooleanProperty(TYPE, processorTag, config, "keep_original", true);
-            return new UriPartsProcessor(processorTag, description, field, targetField, removeIfSuccessful, keepOriginal);
+            boolean ignoreMissing = ConfigurationUtils.readBooleanProperty(TYPE, processorTag, config, "ignore_missing", true);
```
[the remark this answers is not in the document]
Comment (afoucret) [reply modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/UriPartsProcessor.java]: Good catch. It is fixed.

**Comment by afoucret** [comment]
@joegallo You have a good point about handling of the null value.

I did fix your failing test and the behavior should be OK right now:
- `null` value with `ignore_missing` set to `false`  => error is thrown
- `null` value with `ignore_missing` set to `true`  => the doc is left intact

**Comment by afoucret** [comment]
@elasticmachine run elasticsearch-ci/packaging-tests-unix-sample

**Comment by afoucret** [comment]
@elasticmachine run elasticsearch-ci/packaging-tests-unix-sample

**Comment by joegallo** [comment]
@elasticmachine update branch

**Comment by afoucret** [comment]
Rebased the PR.

**Comment by joegallo** [comment]
Argh. Please don't rebase PRs. You can merge `main` in if you want. Technically now I have to review everything again from the beginning as if anything could changed, because indeed it could have.

I'm sure you didn't sneak anything nefarious into this PR by rebasing it, but if I were going to try to sneak something nefarious into somebody else's codebase, rebasing a PR that had already been mostly reviewed would be one of the tools I'd reach for first.

**Review by joegallo — APPROVED** [review]

**Pull request merged**